client/packages/inventory/src/Stocktake/api/api.ts

```diff
@@ -207,3 +211,36 @@ export const getStocktakeQueries = (sdk: Sdk, storeId: string) => ({
     throw new Error('Could not create stocktake');
   },
 });
+
+const getInsertStocktakeLines = (
+  stocktakeId: string,
+  items: { itemId: string; stockLines?: StockLineFragment[] | undefined }[]
+) => {
+  const insertStocktakeLines = [] as InsertStocktakeLineInput[];
+
+  items.forEach(item => {
+    const { itemId, stockLines } = item;
+
+    if (stockLines) {
+      stockLines.forEach(stockLine => {
```

Comment:
I think insert stocktake line endpoint, if stockLineId is specified, it will add all those fields

Reply:
unfortunately they are all null if not specified. have added in `expiryDate` too..

Reply:
oops, I can't remember our discussions with Clemens about this, but possibly, but added an issue: https://github.com/openmsupply/open-msupply/issues/262